### ISSUE TITLE
Modify results structure

### ISSUE THREAD

**Issue by ppinard** [pymontecarlo/pymontecarlo#21]
I would remove the class *Results* and *ResultsContainer*, and simply stores the results inside the *Options*. An *Options* that has not been simulated would simply have no results.

**Comment by ppinard** [comment]
@drix00 said:
I am not sure to understand this point. For me options and results are different and not in the same class. Maybe a simulation (single) class containing the options and results class.

**Comment by ppinard** [comment]
In the current version, _Results_ contains the _options_ with all the different parameters (e.g. different beam energies) and some _ResultsContainer_'s for the results of each simulation (one for each beam energy). 

I agree we should have another class called _Simulation_ that would contains _Options_ (new version with all single parameters, see #18 ), _Results_ and maybe other things like results for a re-analysis or some program information (version, etc.). In short,

```
Simulation:
    options -> Options
    results -> Results
    ...
```

**Comment by drix00** [comment]
Similar to issue #18 for options, we can have result from multiple simulation, i.e., a BSE image or x-ray intensity versus accelerating voltage. In some of my previous code or early design we can have this structure:

- Experiment:
    - Range of options (N simulations to do)
        - Accelerating voltage
        - scan points
        - programs
    - Simulations results (N results)
        - Simulation
            - Options
            - Results
    - Analyze or range of results
        - BSE map
        - x-ray intensity versus accelerating voltage

The problem is the "analyze" and "ranges of options" are not 1 to 1 relationship, but they can share simulation results. The big advantage of HDF5 is you can move or copy any left from one file to another easily. So the user can start a range of options, do some of the simulations, than want to do an specific analysis. So pymontecarlo should be able to read and combine all the three main branch and generate/simulate all the missing simulations. It is not clear how to do it correctly. but we have to keep it in mind when we develop pymontecarlo that we have a triangle relationship between "Range of options", "Simulation", and "Analyze" and they should be able to communicate between each other.

**Comment by ppinard** [comment]
I see the "analyze" section as the task concept (#17). Analysis might be a better name.

Do we need to store the range of options? Should the analysis decide which simulations to use based on those available?

**Comment by ppinard** [comment]
This is mostly implemented in version [1.0](https://github.com/pymontecarlo/pymontecarlo/releases/tag/1.0.0), but the following question remains:

Do we need to store the range of options?

**Comment by ppinard** [comment]
No longer applicable
